server/rest.py

```diff
@@ -151,11 +151,14 @@ def getTile(self, itemId, z, x, y, params):
         except ValueError:
             raise RestException('x, y, and z must be integers', code=400)
 
-        tileSource = self._loadTileSource(itemId)
+        tileSource = self._loadTileSource(itemId, params)
         try:
             tileData = tileSource.getTile(x, y, z)
         except TileSourceException as e:
             raise RestException(e.message, code=404)
 
-        cherrypy.response.headers['Content-Type'] = 'image/jpeg'
+        if tileData[:8] == '\x89PNG\r\n\x26\n':
```

Comment:
I think the TileSource should make the tile encoding type available via a method, instead of peeking into binary data in the API layer.